odoo/addons/base/populate/res_partner.py

```diff
@@ -0,0 +1,160 @@
+            ('type', populate.constant('contact')), # todo add more logic, manage 'invoice', 'delivery', 'other', 'private'
+            ('is_company', populate.iterate([True, False], [0.05, 0.95])),
+            ('_address', address_generator),
+            ('state_id', populate.compute(state_callable)),
+            ('phone', populate.randomize([False, '', '+3212345678', '003212345678', '12345678'])),
+            ('mobile', populate.randomize([False, '', '+32412345678', '0032412345678', '412345678'])),
+            ('title', populate.randomize(self.env['res.partner.title'].search([]).ids)),
+            ('function', populate.randomize(
+                [False, '', 'President of Sales', 'Senior Consultant', 'Product owner', 'Functional Consultant', 'Chief Executive Officer'],
+                [50, 10, 2, 20, 5, 10, 1])),
+            ('tz', populate.randomize([tz for tz in self.env['res.partner']._fields['tz'].get_values(self.env)])),
+            ('website', populate.randomize([False, '', 'http://www.example.com'])),
+            ('credit_limit', populate.randomize(
+                [False, 0, 500, 2500, 5000, 10000],
+                [0.50, 0.30, 0.5, 0.5, 0.5, 0.5])),
+            ('name', populate.compute(name_callable)), # keep after is_company
+            ('ref', populate.randomize([False, '', '{counter}', 'p-{counter}'], [10, 10, 30, 50])),
+            ('industry_id', populate.randomize(
+                [False] + industry_ids,
+                [0.5] + ([0.5/(len(industry_ids) or 1)] * len(industry_ids))))
+        ]
+
+    def _populate_database(self, scale):
+        new = super()._populate_database(scale)
+        # set parent_ids
+        self._populate_set_companies(new)
+        return new
+
+    def _populate_set_companies(self, new):
+        _logger.info('Setting companies')
+        r_company = populate.Random('res.partner+company_has_partners') # 50% change to have partners
+        companies = new.filtered(lambda p: p.is_company and r_company.getrandbits(1))
+        partners = new - companies
+        r_partner = populate.Random('res.partner+partner_has_company')
+        r_company_pick = populate.Random('res.partner+partner_company_pick=')
+        companies_partners = collections.defaultdict(lambda: self.env['res.partner'])
+        for count, partner in enumerate(partners):
+            if bool(r_partner.getrandbits(1)):  # 50% change to have a company
+                companies_partners[r_company_pick.choice(companies)] |= partner
+
+        # batching company write improves performances a lot (~40% faster for total partner creation).
+        for count, (company, partners) in enumerate(companies_partners.items()):
+            if count % 100 == 0:
+                _logger.info('Setting company: %s/%s', count, len(companies))
+            partners.write({'parent_id': company.id})
```

Comment:
I have the feeling that this will generate strange data, like a company in Belgium with contacts in all coutries except Belgium...